File: sections/61_icpc_contest_readiness/problems/a_scoreboard_replay/solution.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int teams, problems, submissions;
    if (!(cin >> teams >> problems >> submissions)) return 0;
    vector<vector<char>> solved(teams + 1, vector<char>(problems + 1));
    vector<vector<int>> wrong(teams + 1, vector<int>(problems + 1));
    vector<int> solved_count(teams + 1);
    vector<long long> penalty(teams + 1);

    while (submissions--) {
        int minute, team, problem;
        char verdict;
        cin >> minute >> team >> problem >> verdict;
        if (solved[team][problem]) continue;
        if (verdict == 'W') {
            ++wrong[team][problem];
        } else {
            solved[team][problem] = true;
            ++solved_count[team];
            penalty[team] += minute + 20LL * wrong[team][problem];
        }
    }

    vector<int> order(teams);
    iota(order.begin(), order.end(), 1);
    sort(order.begin(), order.end(), [&](int a, int b) {
        return tuple{-solved_count[a], penalty[a], a} <
               tuple{-solved_count[b], penalty[b], b};
    });
    for (int team : order) {
        cout << team << ' ' << solved_count[team] << ' ' << penalty[team] << '\n';
    }
}

File: sections/61_icpc_contest_readiness/problems/a_scoreboard_replay/solution.py
import sys


def main() -> None:
    data = sys.stdin.buffer.read().split()
    if not data:
        return
    it = iter(data)
    teams = int(next(it))
    problems = int(next(it))
    submissions = int(next(it))
    solved = [[False] * (problems + 1) for _ in range(teams + 1)]
    wrong = [[0] * (problems + 1) for _ in range(teams + 1)]
    solved_count = [0] * (teams + 1)
    penalty = [0] * (teams + 1)

    for _ in range(submissions):
        minute = int(next(it))
        team = int(next(it))
        problem = int(next(it))
        verdict = next(it)
        if solved[team][problem]:
            continue
        if verdict == b"W":
            wrong[team][problem] += 1
        else:
            solved[team][problem] = True
            solved_count[team] += 1
            penalty[team] += minute + 20 * wrong[team][problem]

    order = sorted(range(1, teams + 1), key=lambda team: (-solved_count[team], penalty[team], team))
    print("\n".join(f"{team} {solved_count[team]} {penalty[team]}" for team in order))


if __name__ == "__main__":
    main()
